community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -753,17 +759,31 @@ public void close( Consumer<PageCursor> headerWriter ) throws IOException
 
     private void close( Header.Writer headerWriter ) throws IOException
     {
-        writer.close();
-
+        writerCheckpointMutex.lock();
```

Comment:
I can't get test to fail locally by removing the lock from close. Seems like test is not testing what it should.
Although, this might not be relevant. The PR suggest we should not checkpoint at all during close https://github.com/neo4j/neo4j/pull/9028